test/framework/src/main/java/org/elasticsearch/cluster/metadata/DataStreamTestHelper.java

```diff
@@ -169,7 +170,8 @@ public static DataStream newInstance(
             lifecycle,
             failureStores.size() > 0,
             failureStores,
-            null
+            null,
+            1
```

Comment:
This should probably be based on the failure stores list size

Reply:
Good catch! Thanks